decomp/semantics/uds.py

```diff
@@ -1000,12 +1044,37 @@ def __init__(self, annotation: Dict[str, Dict[str, Any]]):
                                         if '%%' in edge}
                                 for gname, attrs in self.annotation.items()}
 
-    def __getitem__(self, k):
-        node_attrs = self.node_attributes[k]
-        edge_attrs = self.edge_attributes[k]
+    @classmethod
+    @abstractmethod
```

Comment:
I believe this should be [`abc.abstractclassmethod`](https://docs.python.org/3/library/abc.html#abc.abstractclassmethod).

Reply:
I thought so as well, but this has been deprecated since 3.3, as you can now use `@classmethod` and `@abstractmethod` together, which is apparently the recommended approach.